src/bin/upgrade/main.rs

```diff
@@ -3,25 +3,32 @@
        trivial_numeric_casts, unsafe_code, unstable_features, unused_import_braces,
        unused_qualifications)]
 
+extern crate cargo_metadata;
 extern crate docopt;
 #[macro_use]
 extern crate error_chain;
 #[macro_use]
 extern crate serde_derive;
-extern crate serde_json;
 extern crate toml;
 
+use std::collections::HashMap;
 use std::io::{self, Write};
-use std::process::{self, Command};
+use std::path::Path;
+use std::process;
 
 extern crate cargo_edit;
-use cargo_edit::{get_latest_dependency, Manifest};
+use cargo_edit::{get_latest_dependency, Dependency, Manifest};
 
 mod errors {
     error_chain!{
         links {
             CargoEditLib(::cargo_edit::Error, ::cargo_edit::ErrorKind);
         }
+
+        foreign_links {
+            // cargo-metadata doesn't (yet) export `ErrorKind`)
```

Comment:
trailing `)`

Is there a tracking issue on cargo-metadata to expose its ErrorKind? If not, this comment doesn't seem worth having

Reply:
Oops - yes.

No issue, but https://github.com/oli-obk/cargo_metadata/pull/21 should fix it. I'm hopeful of being able to remove the comment within the next month.